roles/azimuth_caas_operator/defaults/main.yml

```diff
@@ -120,6 +120,37 @@ azimuth_caas_stackhpc_slurm_appliance_extra_vars: >-
     azimuth_caas_stackhpc_slurm_appliance_extra_vars_defaults |
       combine(azimuth_caas_stackhpc_slurm_appliance_extra_vars_overrides)
   }}
+# The tenancy-based access controls for the StackHPC Slurm appliance
+# List of allowed tenancy IDs
+azimuth_caas_stackhpc_slurm_appliance_tenancy_allow_list: "{{ platforms_tenancy_allow_list | default(None) }}"
+# List of denied tenancy IDs
+azimuth_caas_stackhpc_slurm_appliance_tenancy_deny_list: "{{ platforms_tenancy_deny_list | default(None) }}"
+# Regex pattern to allow tenancies by name
+azimuth_caas_stackhpc_slurm_appliance_tenancy_allow_regex: "{{ platforms_tenancy_allow_regex | default(None) }}"
+# Regex pattern to block tenancies by name
+azimuth_caas_stackhpc_slurm_appliance_tenancy_deny_regex: "{{ platforms_tenancy_deny_regex | default(None) }}"
+# The set of key-value annotations to apply to the template
+azimuth_caas_stackhpc_slurm_appliance_tenancy_access_control_annotations: >-
+  {{-
+    {} |
+    combine(
+      { "acl.azimuth.stackhpc.com/allow-list": (azimuth_caas_stackhpc_slurm_appliance_tenancy_allow_list | join(",")) }
+      if azimuth_caas_stackhpc_slurm_appliance_tenancy_allow_list | length > 0
```

Comment:
The `| length` is unnecessary here because an empty list evaluates to `false`. Also, your default is `None` - does that work well with `| length`?

Same applies to all other similar lines.

Reply:
The `None` default did seem to work, but I've removed the `length`'s anyway